src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php

```diff
@@ -136,12 +136,15 @@ public function getSupportedOutputImageFormats(?string $mimeType): array
 
     public function convert(FileVersion $fileVersion, $formatKey, $imageFormat)
     {
-        $imageResource = $this->mediaImageExtractor->extract($this->storage->load($fileVersion->getStorageOptions()));
+        $imageResource = $this->mediaImageExtractor->extract(
+            $this->storage->load($fileVersion->getStorageOptions()),
+            $fileVersion->getMimeType()
```

Comment:
this is set on upload right? how/where do we detect it then? 🙂

Reply:
First the media is uploaded to PHP Server there we detect the mimetype before it is then store over flyssystem in a external storage.